projects/client-api/src/index.js

```diff
@@ -1303,6 +1306,71 @@ export function updateZoom(level) {
 }
 chainList.updateZoom = updateZoom;
 
+/**
+ * Subscribe to label change and exit events
+ * @function selectionUpdates
+ * @param {object} [cache] - An optional cache object (empty object) to use for caching label updates across subscriptions.
+ * @return {Subscription} A {@link Subscription} that can be used to stop reacting to label updates
+ * @example
+ * GraphistryJS(document.getElementById('viz'))
+ *     .pipe(
+ *          selectionUpdates,
+ *          tap((s) => {
+ *                 console.log(s);
+ *          }),
+ *     })
+ *     .subscribe();
+ */
+export function selectionUpdates(g) {
+    console.log('client-api.APISUB', 'selectionUpdates', g, 'exor');
+    if (!(g.subscriptionAPIVersion >= 1)) {
+        throw new Error('selectionUpdates is not available the currently embedded graphistry viz.');
+    }
+
+    const SELECTION_PATH = "workbooks.open.views.current.selection['edge','point']";
+    return g.selectionStream || (g.selectionStream = new BehaviorSubject('value'))
+        .pipe(
+            tap((v) => {
+                console.debug('client-api.APISUB', 'post graphistry-subscribe', 'exor');
+                g.iFrame.contentWindow.postMessage({ type: 'graphistry-subscribe', agent: 'graphistryjs', path: SELECTION_PATH }, '*');
+            }),
+            finalize(() => {
+                console.debug('client-api.APISUB', 'finalize post graphistry-unsubscribe', 'exor');
+                g.iFrame.contentWindow.postMessage({ type: 'graphistry-unsubscribe', agent: 'graphistryjs', path: SELECTION_PATH }, '*');
+            }),
```

Comment:
i'm unsure of relationship with shareReplay: should test

ex: if unsub (ex: close a panel), does it actually unsub (no child pushes?)
ex: if resub on a closed, does it actually resub?

Reply:
Sorted this out, and it works in the storybook example